app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -343,21 +345,29 @@ private static PokemonNameCorrector initCorrectorFromPrefs(PokeInfoCalculator po
 
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
-        if (intent != null && intent.hasExtra(KEY_TRAINER_LEVEL)) {
-            trainerLevel = intent.getIntExtra(KEY_TRAINER_LEVEL, 1);
-            statusBarHeight = intent.getIntExtra(KEY_STATUS_BAR_HEIGHT, 0);
-            batterySaver = intent.getBooleanExtra(KEY_BATTERY_SAVER, false);
-            makeNotification(this);
-            createInfoLayout();
-            createIVButton();
-            createArcPointer();
-            createArcAdjuster();
-            /* Assumes MainActivity initialized ScreenGrabber before starting this service. */
-            if (!batterySaver) {
-                screen = ScreenGrabber.getInstance();
-                startPeriodicScreenScan();
-            } else {
-                screenShotHelper = ScreenShotHelper.start(Pokefly.this);
+        if (intent != null) {
+            if (ACTION_STOP.equals(intent.getAction())) {
+                if (screen != null) {
+                    screen.exit();
```

Comment:
I followed MainActivity example: there the screen is exited and then stopService is called.
I thought that following the same pattern would be better.
If we put that exit call in onDestroy, it will need to be removed from MainActivity or it'll be called twice when the user taps the Stop button in the activity (don't know if that may lead to unexpected behaviors)... Or, at least, this is what I understood given my little knowledge of ScreenGrabber :)